luzer/luzer_args.c

```diff
@@ -0,0 +1,284 @@
+
+	char *corpus_path = NULL;
+	for (int i = 0; i < table_args->argc; i++) {
+		if (strncmp(table_args->argv[i], SEED_CORPUS_PATH_FLAG, strlen(SEED_CORPUS_PATH_FLAG)) == 0) {
+			int corpus_path_len = strlen(table_args->argv[i]) - strlen(SEED_CORPUS_PATH_FLAG);
+			corpus_path = malloc(corpus_path_len * sizeof(char*));
+                        if (!corpus_path) {
+                                return -1;
+                        }
+			memcpy(corpus_path, &table_args->argv[i][strlen(SEED_CORPUS_PATH_FLAG) + 1], corpus_path_len);
+			free(table_args->argv[i]);
+                        table_args->argv[i] = NULL;
+		} else {
+                        char key[strlen(table_args->argv[i]) + 1];
+                        if (sscanf(table_args->argv[i], FLAG_SCANF_FORMAT_KEY, key) == 0) {
+#ifdef DEBUG
+                                DEBUG_PRINT("error get libfuzzer flag in string: %s\n", table_args->argv[i]);
+#endif /* DEBUG */
+                                continue;
+                        }
+
+                        if (is_flag_in_args(cli_args, key)) {
+                                continue;
+                        }
+
+                        if (add_arg_with_mem_allocation(total_args, table_args->argv[i])) {
+                                return -1;
+                        }
+                        total_args->argc++;
+		}
+	}
+
+	for (int i = 1; i < cli_args->argc; i++) {
+                if (add_arg_with_mem_allocation(total_args, cli_args->argv[i])) {
+                        return -1;
+                }
+		total_args->argc++;
+	}
+	if (corpus_path) {
+                if (add_arg_with_mem_allocation(total_args, corpus_path)) {
+                        return -1;
+                }
+		total_args->argc++;
+	}
+
+        if (add_arg_with_mem_allocation(total_args, NULL)) {
+                return -1;
+        }
+
+        if (table_args->argv) {
+                free(table_args->argv);
+        }
+
+        if (cli_args->argv) {
+                free(cli_args->argv);
+        }
+
+	return 0;
+}
+
+NO_SANITIZE static void
+free_args(luzer_args args) {
+        if (!args.argv) {
+                return;
+        }
+        for (int i = 0; i < args.argc; i++) {
+                if (args.argv[i]) {
+                        free(args.argv[i]);
+                }
+        }
+        free(args.argv);
+}
+
+#ifdef DEBUG
+NO_SANITIZE static void
+print_args_with_prefix(luzer_args args, const char* prefix) {
+        for (int i = 0; i < args.argc; i++) {
+                DEBUG_PRINT("libFuzzer %s arg - '%s'\n", prefix, args.argv[i]);
+        }
+}
+#endif /* DEBUG */
+
+NO_SANITIZE int
+luaL_get_fuzz_args(lua_State *L, char ***argv, int *argc) {
+        luzer_args total_args = { .argv = NULL, .argc = 0};
+        luzer_args cli_args = { .argv = NULL, .argc = 0 };
+	luzer_args table_args = { .argv = NULL, .argc = 0 };
+
+        int result = -1;
+        result = luaL_get_args_from_cli(L, &cli_args);
```

Comment:
It is not a self-explained variable name. I propose to rename to something like `retcode`, `rc` or something else.